services/repository-managers/src/main/java/org/sagebionetworks/repo/manager/PermissionsManagerUtils.java

```diff
@@ -43,11 +42,18 @@ public static void validateACLContent(AccessControlList acl, UserInfo userInfo,
 					foundCallerInAcl = true;
 				}
 			}
-			if (ra.getPrincipalId().equals(AuthorizationConstants.BOOTSTRAP_PRINCIPAL.PUBLIC_GROUP.getPrincipalId())
-					&& ra.getAccessType().contains(ACCESS_TYPE.DOWNLOAD)) {
-				throw new InvalidModelException("Public may not have DOWNLOAD access.");
+			// Does not allow ACL for the anonymous user
+			if (ra.getPrincipalId().equals(BOOTSTRAP_PRINCIPAL.ANONYMOUS_USER.getPrincipalId())) {
+				throw new InvalidModelException("Cannot assign permissions to anonymous users");
```

Comment:
"Cannot assign a permission to anonymous.  To share resources with anonymous users, use the PUBLIC groupId: "